pkg/proto/v0/accounts.pb.micro_test.go

```diff
@@ -0,0 +1,927 @@
+	createAccount(t, "user1")
+	createAccount(t, "user2")
+
+	req := &proto.DeleteAccountRequest{Id: getAccount("user1").Id}
+
+	client := service.Client()
+	cl := proto.NewAccountsService("com.owncloud.api.accounts", client)
+
+	resp, err := cl.DeleteAccount(context.Background(), req)
+	checkError(t, err)
+	assert.IsType(t, resp, &empty.Empty{})
+
+	// Check the account doesn't exists anymore
+	accountList, _ := listAccounts(t)
+	assertResponseContainsUser(t, accountList, getAccount("user2"))
+	assertResponseNotContainsUser(t, accountList, getAccount("user1"))
+
+	cleanUp(t)
+}
+
+func TestListGroups(t *testing.T) {
+	req := &proto.ListGroupsRequest{}
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	resp, err := cl.ListGroups(context.Background(), req)
+	checkError(t, err)
+	assert.IsType(t, &proto.ListGroupsResponse{}, resp)
+	assert.Equal(t, len(resp.Groups), 9)
+
+	groups := []string{
+		"sysusers",
+		"users",
+		"sailing-lovers",
+		"violin-haters",
+		"radium-lovers",
+		"polonium-lovers",
+		"quantum-lovers",
+		"philosophy-haters",
+		"physics-lovers",
+	}
+
+	for _, g := range groups {
+		assertResponseContainsGroup(t, resp, getGroup(g))
+	}
+	cleanUp(t)
+}
+
+func TestGetGroups(t *testing.T) {
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	groups := []string{
+		"sysusers",
+		"users",
+		"sailing-lovers",
+		"violin-haters",
+		"radium-lovers",
+		"polonium-lovers",
+		"quantum-lovers",
+		"philosophy-haters",
+		"physics-lovers",
+	}
+
+	for _, g := range groups {
+		group := getGroup(g)
+		req := &proto.GetGroupRequest{Id: group.Id}
+		resp, err := cl.GetGroup(context.Background(), req)
+
+		checkError(t, err)
+		assert.IsType(t, &proto.Group{}, resp)
+		assertGroupsSame(t, group, resp)
+	}
+	cleanUp(t)
+}
+
+func TestCreateGroup(t *testing.T) {
+	group := &proto.Group{Id: "2d58e5ec-842e-498b-8800-61f2ec6f911f", GidNumber: 30002, OnPremisesSamAccountName: "quantum-group", DisplayName: "Quantum Group", Members: []*proto.Account{
+		{Id: "4c510ada-c86b-4815-8820-42cdf82c3d51"}, // einstein
+		{Id: "932b4540-8d16-481e-8ef4-588e4b6b151c"}, // feynman
+	}}
+
+	res, err := createGroup(t, group)
+	checkError(t, err)
+
+	assert.IsType(t, &proto.Group{}, res)
+
+	// The response is Empty *proto.Group
+	// assert.Empty(t, res)
+
+	groupsResponse := listGroups(t)
+	assertResponseContainsGroup(t, groupsResponse, group)
+	cleanUp(t)
+}
+
+func TestGetGroupInvalidID(t *testing.T) {
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	req := &proto.GetGroupRequest{Id: "42"}
+	resp, err := cl.GetGroup(context.Background(), req)
+
+	assert.IsType(t, &proto.Group{}, resp)
+	assert.Empty(t, resp)
+	assert.Error(t, err)
+	assert.Equal(t, "{\"id\":\".\",\"code\":404,\"detail\":\"could not read group: open accounts-store/groups/42: no such file or directory\",\"status\":\"Not Found\"}", err.Error())
```

Comment:
can we unmarshal this string and compare the JSON object?

Reply:
The server returns this as a string so I don't think that will be really necessary